File: pdlearning/src/web-apps/src/libraries/domain-api/src/lna-form/models/form-with-questions.model.ts
import { IStandaloneSurveyModel, StandaloneSurveyModel } from './lna-form.model';
import { ISurveyQuestionModel, SurveyQuestionModel } from './form-question.model';

import { FormSection } from '../../form-section/models/form-section';
import { SurveyAnswerModel } from './form-answer.model';

export interface ISurveyWithQuestionsModel {
  form: IStandaloneSurveyModel;
  formQuestions: ISurveyQuestionModel[];
  formSections: FormSection[];
}
export class SurveyWithQuestionsModel implements ISurveyWithQuestionsModel {
  public form: StandaloneSurveyModel = new StandaloneSurveyModel();
  public formQuestions: SurveyQuestionModel[] = [];
  public formSections: FormSection[] = [];

  constructor(data?: ISurveyWithQuestionsModel) {
    if (data != null) {
      this.form = new StandaloneSurveyModel(data.form);
      this.formQuestions = data.formQuestions.map(_ => new SurveyQuestionModel(_));
      this.formSections = data.formSections.map(_ => new FormSection(_));
    }
  }
}

export interface SurveyQuestionWithAnswerModel {
  formQuestion: SurveyQuestionModel[];
  formAnswer: SurveyAnswerModel;
}

File: pdlearning/src/microservices/lnaform-microservice/Api/Application/Models/FormWithQuestionsModel.cs
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Microservice.LnaForm.Domain.Entities;
using FormEntity = Microservice.LnaForm.Domain.Entities.Form;

namespace Microservice.LnaForm.Application.Models
{
    public class FormWithQuestionsModel
    {
        public FormWithQuestionsModel()
        {
            // Do not remove this constructor. It is being used by JsonSerializer.Deserialize
        }

        public FormWithQuestionsModel(FormEntity formEntity, IEnumerable<FormQuestion> formQuestionEntities, IEnumerable<FormSection> formSectionEntities, bool canUnpublishStandalone)
        {
            Form = new FormModel(formEntity, canUnpublishStandalone);
            FormQuestions = formQuestionEntities.Select(question =>
            {
                var questionModel = new FormQuestionModel(question);
                questionModel.QuestionTitle = HttpUtility.HtmlDecode(questionModel.QuestionTitle);
                return questionModel;
            });
            FormSections = formSectionEntities.Select(section => new FormSectionModel(section));
        }

        public FormModel Form { get; set; }

        public IEnumerable<FormQuestionModel> FormQuestions { get; set; }

        public IEnumerable<FormSectionModel> FormSections { get; set; }
    }
}
